### PR TITLE
Reorganize demos to give a consistent and homogeneus structure.

### PR DESCRIPTION
**Pull request by jmoreira-valory**
## Proposed changes

Reorganize demos to give a consistent and homogeneus structure. The proposed structure for each demo in the repository is as follows. 

- Introduction
- Architecture (include diagram)
- Running the Demo: All demo services should be available in the Registry / IPFS.
- Interacting with the demo: How the user can see/modify the outcome of the demo. (*)
- Technical details (*)
- FSM specification (*)

Not all the sections are mandatory (e.g., if one section marked with (*) is too small, can be integrated within another section.)

Note that some of the services are not already available in the registry. A disclaimer note has been added until they are fully available.

## Fixes

If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [ ] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have locally run services that could be impacted and they do not present failures derived from my changes

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: docs/counter_example.md [hunk 1]
```diff
@@ -1,53 +1,102 @@
...
+    ```bash
+    mkdir my_demo
+    cd my_demo
+    ```
+
+2. Within the workspace folder, setup the environment:
+    ```bash
+    export VERSION=0.1.0
+    export OPEN_AEA_IPFS_ADDR="/dns/registry.autonolas.tech/tcp/443/https"
+    touch Pipfile && pipenv --python 3.10 && pipenv shell
```
Comment (dvilelaf) [inline docs/counter_example.md:38]: Have you run this? Make sure that they follow the quickstart steps. I think there is a missing autonomy init needed for those that never had run anything before.

File: docs/hello_world_agent_service.md [hunk 2]
```diff
@@ -191,7 +191,7 @@ As a summary, find below an image which shows the main components of the agent a
 </figure>
 
 ## Coding the Hello World! Agent Service: A Primer
-So far, we have given a conceptual description of the Hello World {{agent_service}}. As we have seen, there are a number of components that a developer needs to focus on in order to fully define the service.
+So far, we have given a conceptual description of the Hello World agent service. As we have seen, there are a number of components that a developer needs to focus on in order to fully define the service.
```
Comment (dvilelaf) [inline docs/hello_world_agent_service.md:194]: Why are we removing this variable?

File: docs/price_oracle_intro.md [hunk 3]
```diff
@@ -1,322 +1,222 @@
...
+    ```bash
+    mkdir my_demo
+    cd my_demo
+    ```
+
+2. Within the workspace folder, setup the environment:
+    ```bash
+    export VERSION=0.1.0
+    export OPEN_AEA_IPFS_ADDR="/dns/registry.autonolas.tech/tcp/443/https"
+    touch Pipfile && pipenv --python 3.10 && pipenv shell
```
Comment (dvilelaf) [inline docs/price_oracle_intro.md:55]: Same as above

File: docs/price_oracle_intro.md [hunk 4]
```diff
@@ -1,322 +1,222 @@
...
-### The `SafeDeploymentAbciApp`
-
-This `AbciApp` implements the deployments of a Gnosis safe contract, which is
-a multisig smart contract wallet that requires a minimum number of people to
-approve a transaction before it can occur. This assures that no single agent
-can compromise the funds contained in it.
-
-0. `RandomnessSafeRound` <br/>
-   Some randomness is retrieved to be used in a keeper agent selection. In
-   particular, agents individually request the latest random number from
```
Comment (dvilelaf) [inline docs/price_oracle_intro.md:107]: Has this been completely removed or just moved?

**Review by dvilelaf — COMMENTED** [review]
There's some failing tests. Also some comments.

File: docs/hello_world_agent_service.md (hunk 2 above)
Comment (jmoreira-valory) [reply docs/hello_world_agent_service.md:194]: I try to follow and try to enforce the criteria of using variables for proper names, e.g., {{open_autonomy}}, which are frequently repeated across the documentation.

File: docs/price_oracle_intro.md (hunk 4 above)
Comment (jmoreira-valory) [reply docs/price_oracle_intro.md:107]: This has been moved to the technical details section (in another file). The overall description of demos should follow:
Intro - Architecture - Running - Interacting - Technical Details - (FSMs)
In this case, Technical details and FSMs are in separate files.

File: docs/simple_abci.md [hunk 5]
```diff
@@ -27,6 +29,87 @@ The agents are connected to the remote service [DRAND](https://drand.love) throu
   <figcaption>Simple ABCI architecture with four agents</figcaption>
 </figure>
 
+{% include 'requirements.md' %}
+
+## Running the Demo
+
+!!! note
```
Comment (jmoreira-valory) [inline docs/simple_abci.md:36]: @DavidMinarsch please check if you are happy with this non-ideal approach.

File: docs/counter_example.md [hunk 6]
```diff
@@ -1,53 +1,102 @@
-# ABCI Counter with AEAs
+# Counter Agent Service Demo
```
Comment (DavidMinarsch) [inline docs/counter_example.md:1]: I don't think this is a service. It is just a degenerate version of a service. So I would not use the term agent service here

File: docs/counter_example.md [hunk 7]
```diff
@@ -1,53 +1,102 @@
...
+
+1. Open a terminal and create a workspace folder, e.g.,
+    ```bash
+    mkdir my_demo
+    cd my_demo
+    ```
+
+2. Within the workspace folder, setup the environment:
+    ```bash
+    export VERSION=0.1.0
```
Comment (DavidMinarsch) [inline docs/counter_example.md:36]: What's that needed for?

File: docs/counter_example.md (hunk 6 above)
Comment (DavidMinarsch) [reply docs/counter_example.md:1]: I was wrong.., there is a service for this ...interesting

File: docs/price_oracle_intro.md [hunk 8]
```diff
@@ -1,322 +1,222 @@
...
+
+1. Open a terminal and create a workspace folder, e.g.,
+    ```bash
+    mkdir my_demo
+    cd my_demo
+    ```
+
+2. Within the workspace folder, setup the environment:
+    ```bash
+    export VERSION=0.1.0
```
Comment (DavidMinarsch) [inline docs/price_oracle_intro.md:53]: ?

File: docs/simple_abci.md (hunk 5 above)
Comment (DavidMinarsch) [reply docs/simple_abci.md:36]: @jmoreira-valory no @dvilelaf should open a PR on top here adding the agent service and then you can finish your PR once merged into this PR

File: docs/counter_example.md (hunk 1 above)
Comment (DavidMinarsch) [reply docs/counter_example.md:38]: We should assume that the setup has happened in the quickstart. So the demos should not contain these setup steps at all. Just say, make sure you're set up...

File: docs/price_oracle_intro.md (hunk 8 above)
Comment (jmoreira-valory) [reply docs/price_oracle_intro.md:53]: Removed

File: docs/counter_example.md (hunk 7 above)
Comment (jmoreira-valory) [reply docs/counter_example.md:36]: Removed.

File: docs/counter_example.md (hunk 6 above)
Comment (jmoreira-valory) [reply docs/counter_example.md:1]: It seems that it is a non-FSM based service.

File: docs/counter_example.md (hunk 1 above)
Comment (jmoreira-valory) [reply docs/counter_example.md]: Fixed.

**Review by DavidMinarsch — APPROVED** [review]

**Pull request merged**